crucible/src/Lang/Crucible/Simulator/ExecutionTree.hs

```diff
@@ -1075,7 +1076,26 @@ ctxSymInterface = to (\ctx ->
 functionBindings :: Lens' (SimContext p sym ext) (FunctionBindings p sym ext)
 functionBindings = lens _functionBindings (\s v -> s { _functionBindings = v })
 
--- | Access the custom user-state inside the 'SimContext'.
+-- | Custom state inside the 'SimContext'.
+--
+-- Crucible itself is entirely polymorphic over @p@, downstream applications can
+-- instantiate it to any sort of state that they would like to associate with a
+-- 'SimContext'.
+--
+-- For example, Macaw-based applications can instantiate this to
```

Comment:
Perhaps this should be `macaw-symbolic` instead of Macaw, as the core of Macaw doesn't use Crucible.

Also, consider linking to `macaw-symbolic` here, as readers may not be familiar with what it is.